### PR TITLE
[INFINITY-2621] spark<->kafka ci test

### PR DESCRIPTION
**Pull request by ArtRand**
This PR adds a Kafka/Spark inter-service test that does the following:
1. Installs a Kerberized Kafka cluster using `dcos-commons` test tooling
2. Starts `KafkaProducer` a Spark job that pushes small sentences into a topic `top1`
3. Starts `KafkaConsumer` that reads from this topic and stops when 48 words have been processed.
4. Upon successful finish of (3) tears the whole test apparatus down. 

Other changes in this PR include: 
1. Changing the JCE/JVM to align with `dcos-commons` (8u131)
2. Add JAAS file to `resources`.

### CODE REVIEW

**Comment by ArtRand** [comment]
retest this please

File: tests/test_kafka.py [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+@pytest.mark.skipif(not utils.kafka_enabled(), reason='KAFKA_ENABLED is false')
+def test_spark_and_kafka():
+    def producer_launched():
+        return utils.streaming_job_launched(PRODUCER_SERVICE_NAME)
+
+    def producer_started():
+        return utils.streaming_job_running(PRODUCER_SERVICE_NAME)
+
+    def kafka_broker_dns():
+        cmd = "dcos {package_name} --name={service_name} endpoints broker".format(
```
Comment (susanxhuynh) [inline tests/test_kafka.py]: nit: you could use this helper function instead: https://github.com/mesosphere/spark-build/blob/master/testing/sdk_cmd.py#L35

File: tests/test_kafka.py [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+    if KERBERIZED_KAFKA:
+        jaas_path = os.path.join(THIS_DIR, "resources", "spark-kafka-client-jaas.conf")
+        s3.upload_file(jaas_path)
+        jaas_uri = s3.s3_http_url("spark-kafka-client-jaas.conf")
+        uris += ",{}".format(jaas_uri)
+
+    common_args = [
+        "--conf", "spark.mesos.containerizer=mesos",
+        "--conf", "spark.scheduler.maxRegisteredResourcesWaitingTime=2400s",
+        "--conf", "spark.scheduler.minRegisteredResourcesRatio=1.0",
```
Comment (susanxhuynh) [inline tests/test_kafka.py]: What are these two configs for ("min/maxRegisteredResources")? Are they required? ^^

File: tests/test_kafka.py [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+    if KERBERIZED_KAFKA:
+        producer_config += kerberos_args
+
+    producer_id = utils.submit_job(app_url=utils._scala_test_jar_url(),
+                                   app_args=producer_args,
+                                   app_name="/spark",
+                                   args=producer_config)
+
+    shakedown.wait_for(lambda: producer_launched(), ignore_exceptions=False, timeout_seconds=600)
+    shakedown.wait_for(lambda: producer_started(), ignore_exceptions=False, timeout_seconds=600)
```
Comment (susanxhuynh) [inline tests/test_kafka.py]: Nit: you could use this method instead: https://github.com/mesosphere/spark-build/blob/master/tests/utils.py#L196

**Comment by skonto** [comment]
@ArtRand connection to zk is kerberized?

File: tests/jobs/scala/src/main/scala/KafkaJobs.scala [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+
+import org.apache.spark.{SparkConf, SparkContext}
+import org.apache.kafka.clients.producer.{KafkaProducer, ProducerConfig, ProducerRecord}
+import org.apache.spark.storage.StorageLevel
+import org.apache.kafka.common.serialization.StringDeserializer
+import org.apache.spark.rdd.RDD
+import org.apache.spark.streaming.kafka010._
+import org.apache.spark.util.LongAccumulator
+import org.apache.spark.streaming.{Seconds, StreamingContext, Time}
+import org.apache.spark.streaming.receiver._
```
Comment (skonto) [inline tests/jobs/scala/src/main/scala/KafkaJobs.scala]: picky: try use imports in alphabetical order.

File: tests/jobs/scala/src/main/scala/KafkaJobs.scala [hunk 5]
```diff
@@ -0,0 +1,162 @@
+import java.util
+
+import org.apache.spark.{SparkConf, SparkContext}
+import org.apache.kafka.clients.producer.{KafkaProducer, ProducerConfig, ProducerRecord}
```
Comment (skonto) [inline tests/jobs/scala/src/main/scala/KafkaJobs.scala]: [warn] /home/ubuntu/repos/OSS/spark-build/tests/jobs/scala/src/main/scala/KafkaJobs.scala:4: imported `KafkaProducer' is permanently hidden by definition of object KafkaProducer

File: tests/jobs/scala/build.sbt [hunk 6]
```diff
@@ -3,8 +3,11 @@ lazy val root = (project in file("."))
     name := "dcos-spark-scala-tests",
     scalaVersion := "2.11.8",
     libraryDependencies ++= Seq(
-      "org.apache.spark" %% "spark-core" % "2.1.0" % "provided",
+      "org.apache.spark" %% "spark-core" % "2.2.0" % "provided",
+      "org.apache.spark" % "spark-streaming_2.11" % "2.2.0" % "provided",
+      "org.apache.spark" % "spark-streaming-kafka-0-10_2.11" % "2.2.0",
       "org.apache.hadoop" % "hadoop-aws" % "2.6.0",
+      "org.apache.kafka" % "kafka_2.11" % "0.8.2.0",
       "com.github.scopt" %% "scopt" % "3.7.0"
```
Comment (skonto) [inline tests/jobs/scala/build.sbt]: You could use several checks:
scalacOptions ++= Seq(
  "-deprecation",
  "-encoding", "UTF-8",
  "-feature",
  "-language:existentials",
  "-language:higherKinds",
  "-language:implicitConversions",
  "-language:experimental.macros",
  "-language:postfixOps",
  "-unchecked",
  "-Xlint",
  "-Yinline-warnings",
  "-Yno-adapted-args",
  "-Ywarn-dead-code",
  "-Ywarn-numeric-widen",
  "-Ywarn-unused-import",
  "-Xfuture")

File: tests/jobs/scala/src/main/scala/KafkaJobs.scala (hunk 5 above)
Comment (skonto) [reply tests/jobs/scala/src/main/scala/KafkaJobs.scala]: Also please use a package for the jobs in that folder like: dco.spark.test

File: tests/jobs/scala/src/main/scala/KafkaJobs.scala [hunk 7]
```diff
@@ -0,0 +1,162 @@
...
+      synchronized {
+        if (instance == null) {
+          instance = sc.longAccumulator("WordAccumulator")
+        }
+      }
+    }
+    instance
+  }
+}
+
```
Comment (skonto) [inline tests/jobs/scala/src/main/scala/KafkaJobs.scala]: picky: space

File: docker/Dockerfile [hunk 8]
```diff
@@ -45,14 +46,15 @@ RUN mkdir -p /opt/mesosphere/ && \
...
-    curl -L -O https://downloads.mesosphere.com/java/jre-8u152-linux-x64-jce-unlimited.tar.gz && \
-    tar zxf jre-8u152-linux-x64-jce-unlimited.tar.gz && \
-    rm jre-8u152-linux-x64-jce-unlimited.tar.gz
+    curl -L -O https://downloads.mesosphere.com/java/jre-8u131-linux-x64-jce-unlimited.tar.gz && \
+    tar zxf jre-8u131-linux-x64-jce-unlimited.tar.gz && \
+    rm jre-8u131-linux-x64-jce-unlimited.tar.gz
 
 ENV BOOTSTRAP /opt/mesosphere/bootstrap
-ENV JAVA_HOME /usr/lib/jvm/jre1.8.0_152
+ENV JAVA_HOME /usr/lib/jvm/jre1.8.0_131
```
Comment (skonto) [inline docker/Dockerfile]: should we update to 151?

File: tests/jobs/scala/src/main/scala/KafkaJobs.scala [hunk 9]
```diff
@@ -0,0 +1,162 @@
...
+
+object KafkaConsumer {
+  def main(args: Array[String]): Unit = {
+    if (args.length != 4) {
+      throw new IllegalArgumentException("USAGE: <brokerlist> <topic> <stop_at> <use kerberos? [true/false]>")
+    }
+
+    val Array(brokers, topic, stopcount, kerberized) = args
+
```
Comment (skonto) [inline tests/jobs/scala/src/main/scala/KafkaJobs.scala]: picky: space

File: tests/test_kafka.py (hunk 2 above)
Comment (ArtRand) [reply tests/test_kafka.py]: They're not required, but it makes it so the driver will not start sending tasks until it has all of it's executors registered. I found this made running the job more stable.

File: docker/Dockerfile (hunk 8 above)
Comment (ArtRand) [reply docker/Dockerfile]: 151 was broken! 131 was the answer, also this is what we're using (currently at least) in the SDK services.

File: tests/jobs/scala/build.sbt (hunk 6 above)
Comment (ArtRand) [reply tests/jobs/scala/build.sbt]: good tip

File: tests/test_kafka.py (hunk 2 above)
Comment (susanxhuynh) [reply tests/test_kafka.py]: Ok

File: docker/Dockerfile (hunk 8 above)
Comment (skonto) [reply docker/Dockerfile]: ok good to know!

**Comment by skonto** [comment]
LGTM. I tried the PR it run fine.

**Pull request merged**